app/main.py

```diff
@@ -1,13 +1,33 @@
 def format_linter_error(error: dict) -> dict:
-    # write your code here
+    return {
+        "line": error.get("line_number"),
+        "column": error.get("column_number"),
+        "message": error.get("text"),
+        "name": error.get("code"),
+        "source": "flake8"
+    }
     pass
 
-
 def format_single_linter_file(file_path: str, errors: list) -> dict:
-    # write your code here
+    return {"errors":
+        [{format_linter_error(error)} for error in errors],
+    "path": file_path,
+    "status": "passed"
+    if errors == []
+    else "failed"
+    }
     pass
 
-
 def format_linter_report(linter_report: dict) -> list:
-    # write your code here
+    return [
+            {"errors": [], "path": list(linter_report)[0], "status": "passed"},
+            {
+                "errors": [
+                    {format_single_linter_file(linter_report["errors"])}
+                    for errors in linter_report],
+                "path": list(linter_report)[1],
+                "status": "passed"
+                if "errors" == []
+                else "failed"},
+        ]
     pass
```

Comment:
The `pass` statement is not needed here. It is used as a placeholder for future code. But since you have already implemented the function, you can safely remove it.